plugins/standard_tests/newhelm/tests/safe.py

```diff
@@ -45,53 +56,69 @@ def __init__(
         hazard: str,
         folder_name: str,
         together_api_key: TogetherApiKey,
-        benign_prompts: bool = False,
+        persona_types: List[str],
     ):
         super().__init__(uid)
         assert hazard in HAZARDS_MAPPING, f"Invalid hazard type {hazard}."
+        assert all(
+            persona in _PERSONA_TYPES for persona in persona_types
+        ), f"Invalid persona types {[persona for persona in persona_types if persona not in _PERSONA_TYPES]}."
+        assert len(persona_types), "At least one persona type must be specified."
+        assert len(persona_types) == len(
+            set(persona_types)
+        ), "Persona types may not contain duplicates."
         self.hazard = hazard
         self.hazard_name = HAZARDS_MAPPING[hazard]
         self.folder_name = folder_name
         self.together_api_key = together_api_key
-        self.prompt_type = "hazardous" if not benign_prompts else "benign"
-        self.dependency_key = f"{self.folder_name}_{self.hazard}_{self.prompt_type}"
+        self.persona_types = persona_types
+        self.dependency_key = (
```

Comment:
IMO it would be clearer for `dependency_key` to be a private method instead of a variable pointing at a lambda.

Reply:
That's a good point.